src/modules/launcher/Wox.Infrastructure/Image/ImageLoader.cs

```diff
@@ -43,20 +44,48 @@ public static class ImageLoader
             ".ico",
         };
 
+        // Checks whether it is a valid PNG by checking the 8 bytes at the beginning of the file.
+        public static bool IsValidPngSignature(string filePath)
+        {
+            byte[] pngSignature = { 137, 80, 78, 71, 13, 10, 26, 10 };
+            byte[] buffer = new byte[8];
+
+            using FileStream fs = new(filePath, FileMode.Open, FileAccess.Read);
+            return fs.Read(buffer, 0, buffer.Length) == buffer.Length && pngSignature.SequenceEqual(buffer);
+        }
+
         public static void Initialize(Theme theme)
         {
             _hashGenerator = new ImageHashGenerator();
 
             foreach (var icon in new[] { Constant.ErrorIcon, Constant.LightThemedErrorIcon })
             {
-                BitmapImage bmi = new BitmapImage();
-                bmi.BeginInit();
-                bmi.UriSource = new Uri(icon);
-                bmi.CacheOption = BitmapCacheOption.OnLoad;
-                bmi.EndInit();
-                ImageSource img = bmi;
-                img.Freeze();
-                ImageCache[icon] = img;
+                var uri = new Uri(icon);
+
+                try
+                {
+                    if (File.Exists(uri.LocalPath) && IsValidPngSignature(uri.LocalPath))
+                    {
+                        Log.Info($"Image file '{icon}' is a valid PNG.", MethodBase.GetCurrentMethod().DeclaringType);
```

Comment:
Only `debug` or remove it. We should not spam the log file.

Reply:
Done.